containers/src/Kokkos_DynRankView.hpp

```diff
@@ -2191,21 +2191,51 @@ inline std::enable_if_t<Impl::is_view_ctor_property<I>::value> resize(
 
 /** \brief  Resize a view with copying old data to new data at the corresponding
  * indices. */
-template <class... I, class T, class... P>
+template <class... ViewCtorArgs, class T, class... P>
 inline void impl_realloc(DynRankView<T, P...>& v, const size_t n0,
                          const size_t n1, const size_t n2, const size_t n3,
                          const size_t n4, const size_t n5, const size_t n6,
-                         const size_t n7, const I&... arg_prop) {
-  using drview_type = DynRankView<T, P...>;
+                         const size_t n7,
+                         const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop) {
+  using drview_type      = DynRankView<T, P...>;
+  using alloc_prop_input = Impl::ViewCtorProp<ViewCtorArgs...>;
 
   static_assert(Kokkos::ViewTraits<T, P...>::is_managed,
                 "Can only realloc managed views");
+  static_assert(!alloc_prop_input::has_label,
+                "The view constructor arguments passed to Kokkos::realloc must "
+                "not include a label!");
+  static_assert(!alloc_prop_input::has_pointer,
+                "The view constructor arguments passed to Kokkos::realloc must "
+                "not include a pointer!");
+  static_assert(!alloc_prop_input::has_memory_space,
+                "The view constructor arguments passed to Kokkos::realloc must "
+                "not include a memory space instance!");
 
   const std::string label = v.label();
 
   v = drview_type();  // Deallocate first, if the only view to allocation
```

Comment:
Not a flaw in this PR, but the global `Kokkos::fence()` implied by a deallocation does unfortunately undermine some of the goal of #4793.

Mentioning as much for the back-link over there as anything else.

Reply:
The same applies for `View` as well

Reply:
I'm very happy to discuss that again. I'm in favor in restricting to execution spaces that can actually access the memory space.